server/src/termdb.matrix.js

```diff
@@ -257,7 +257,7 @@ export async function getSampleData_dictionaryTerms_termdb(q, termWrappers) {
 		if (limitMutatedSamples && !limitMutatedSamples.has(sample)) continue // this sample is not mutated for given genes
 		if (!samples[sample]) samples[sample] = { sample }
 		const tw = twByTermId[term_id]
-		samples[sample][term_id] = { key, value }
+		samples[sample][term_id] = { key, value: key }
```

Comment:
@siosonel  needs to evaluate if this is a valid fix for the [groupsetting legend issue](http://localhost:3000/?noheader=1&mass={%22dslabel%22:%22PNET%22,%22genome%22:%22hg19%22,%22plots%22:[{%22chartType%22:%22matrix%22,%22termgroups%22:[{%22lst%22:[{%22term%22:{%22name%22:%22TP53%22,%22type%22:%22geneVariant%22}},{%22id%22:%22Gender%22},{%22id%22:%22TSNE%20Category%22,%22q%22:{%22isAtomic%22:true,%22hiddenValues%22:{},%22type%22:%22custom-groupset%22,%22groupsetting%22:{%22inuse%22:true,%22customset%22:{%22groups%22:[{%22values%22:[{%22key%22:%22HGNET_BCOR%22,%22label%22:%22HGNET_BCOR%22},{%22key%22:%22CNS_SARC_DICER%22,%22label%22:%22CNS_SARC_DICER%22},{%22key%22:%22CNS_NB_FOXR2%22,%22label%22:%22CNS_NB_FOXR2%22},{%22key%22:%22Not_available%22,%22label%22:%22Not_available%22},{%22key%22:%22CNS_SARC_CIC%22,%22label%22:%22CNS_SARC_CIC%22},{%22key%22:%22CNS_ET_NEC/NOS%22,%22label%22:%22CNS_ET_NEC/NOS%22},{%22key%22:%22PXA%22,%22label%22:%22PXA%22},{%22key%22:%22EWS%22,%22label%22:%22EWS%22},{%22key%22:%22CPC%22,%22label%22:%22CPC%22}],%22name%22:%221%22,%22type%22:%22values%22},{%22values%22:[{%22key%22:%22HGG%22,%22label%22:%22HGG%22},{%22key%22:%22ETMR%22,%22label%22:%22ETMR%22}],%22name%22:%222%22,%22type%22:%22values%22}]}}}}]}]}]})

Reply:
Yes, I am not sure if this fix should be on the server side or just on the legend. Lets wait for the input from @siosonel .

Reply:
This `value: key` change breaks survival tests, http://localhost:3000/testrun.html?name=survival.integration. There are some backend datapoints that require the key and value to mean different things, including numeric bin label/continuous values, or survival TTE/exit code. 

I pushed a fix to only change the legend data processing.

@airenzp Please remember to run [all integration tests locally](http://localhost:3000/testrun.html?name=*.integration) when opening a PR.